test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -2647,14 +2613,63 @@ public void run() {
 
             long end = System.nanoTime();
             if ((end - start) < 500000000)
-                fail("Test failed: waitFor didn't take long enough");
+                fail("Test failed: waitFor didn't take long enough (" + (end - start) + "ns)");
 
             p.destroy();
 
             p.waitFor(1000, TimeUnit.MILLISECONDS);
         } catch (Throwable t) { unexpected(t); }
     }
 
+    // Path to native executables, if any
+    private static final String TEST_NATIVEPATH = System.getProperty("test.nativepath");
+
+    // Path where "sleep" program may be found" or null
+    private static final Path SLEEP_PATH = initSleepPath();
+
+    /**
+     * Compute the Path to a sleep executable.
+     * @return a Path to sleep or BasicSleep(.exe) or null if none
+     */
+    private static Path initSleepPath() {
+        if (Windows.is() && TEST_NATIVEPATH != null) {
+            // exeBasicSleep is equivalent to sleep on Unix
+            Path exePath = Path.of(TEST_NATIVEPATH).resolve("BasicSleep.exe");
+            if (exePath.toFile().canExecute()) {
+                return exePath;
+            }
+        }
+
+        Path exePath = Path.of("/bin").resolve("sleep");
+        if (exePath.toFile().canExecute()) {
+            return exePath;
+        }
```

Comment:
Not sure why this is so elaborate when elsewhere in the test we just assume `/usr/bin/env` exists?